cockatrice/src/messagelogwidget.cpp

```diff
@@ -219,278 +116,379 @@ QPair<QString, QString> MessageLogWidget::getFromStr(CardZone *zone, QString car
+                                            .arg(startCard)
+                                            .arg(sanitizeHtml(targetPlayer->getName()))
+                                            .arg(targetCard));
+        } else if (player == targetPlayer) {
+            str = tr("%1 points from %2's %3 to their own %4.");
+            appendHtmlServerMessage(str.arg(sanitizeHtml(player->getName()))
+                                            .arg(sanitizeHtml(startPlayer->getName()))
+                                            .arg(startCard)
+                                            .arg(targetCard));
+        } else {
+            str = tr("%1 points from %2's %3 to %4's %5.");
+            appendHtmlServerMessage(str.arg(sanitizeHtml(player->getName()))
+                                            .arg(sanitizeHtml(startPlayer->getName()))
+                                            .arg(startCard)
+                                            .arg(sanitizeHtml(targetPlayer->getName()))
+                                            .arg(targetCard));
+        }
+    }
+}
+
+void MessageLogWidget::logCreateToken(Player *player, QString cardName, QString pt)
+{
+    appendHtmlServerMessage(tr("%1 creates token: %2%3.")
+                                    .arg(sanitizeHtml(player->getName()))
+                                    .arg(cardLink(cardName))
+                                    .arg(pt.isEmpty() ? QString() : QString(" (%1)")
+                                            .arg(sanitizeHtml(pt))));
+}
+
+void MessageLogWidget::logDeckSelect(Player *player, QString deckHash, int sideboardSize)
+{
+    if (sideboardSize < 0)
+        appendHtmlServerMessage(tr("%1 has loaded a deck (%2).")
+                                        .arg(sanitizeHtml(player->getName()))
+                                        .arg(deckHash));
+    else
+        appendHtmlServerMessage(tr("%1 has loaded a deck with %2 sideboard cards (%3).")
+                                        .arg(sanitizeHtml(player->getName()))
+                                        .arg("<font color=\"blue\">" + QString::number(sideboardSize) + "</font>")
+                                        .arg(deckHash));
+}
+
+void MessageLogWidget::logDestroyCard(Player *player, QString cardName)
+{
+    appendHtmlServerMessage(tr("%1 destroys %2.")
+                                    .arg(sanitizeHtml(player->getName()))
+                                    .arg(cardLink(cardName)));
+}
+
+void MessageLogWidget::logDoMoveCard(LogMoveCard &lmc)
+{
+    QString startZone = lmc.startZone->getName();
+    QString targetZone = lmc.targetZone->getName();
+    bool ownerChanged = lmc.startZone->getPlayer() != lmc.targetZone->getPlayer();
+
+    // do not log if moved within the same zone
+    if ((startZone == tableConstant() && targetZone == tableConstant() && !ownerChanged)
+            || (startZone == handConstant() && targetZone == handConstant())
```

Comment:
owner changed still might be a thing here, as I _could_ drag a card from my hand to my opponents hand (no idea why, but some games might do that?)